edk2toollib/uefi/fmp_auth_header.py

```diff
@@ -0,0 +1,195 @@
+    #   /// be EFI_CERT_TYPE_PKCS7_GUID.
+    #   ///
+    #   WIN_CERTIFICATE_UEFI_GUID               AuthInfo;
+    # } EFI_FIRMWARE_IMAGE_AUTHENTICATION;
+    #
+    # ///
+    # /// Certificate which encapsulates a GUID-specific digital signature
+    # ///
+    # typedef struct {
+    #   ///
+    #   /// This is the standard WIN_CERTIFICATE header, where
+    #   /// wCertificateType is set to WIN_CERT_TYPE_EFI_GUID.
+    #   ///
+    #   WIN_CERTIFICATE   Hdr;
+    #   ///
+    #   /// This is the unique id which determines the
+    #   /// format of the CertData. .
+    #   ///
+    #   EFI_GUID          CertType;
+    #   ///
+    #   /// The following is the certificate data. The format of
+    #   /// the data is determined by the CertType.
+    #   /// If CertType is EFI_CERT_TYPE_RSA2048_SHA256_GUID,
+    #   /// the CertData will be EFI_CERT_BLOCK_RSA_2048_SHA256 structure.
+    #   ///
+    #   UINT8            CertData[1];
+    # } WIN_CERTIFICATE_UEFI_GUID;
+    #
+    # ///
+    # /// The WIN_CERTIFICATE structure is part of the PE/COFF specification.
+    # ///
+    # typedef struct {
+    #   ///
+    #   /// The length of the entire certificate,
+    #   /// including the length of the header, in bytes.
+    #   ///
+    #   UINT32  dwLength;
+    #   ///
+    #   /// The revision level of the WIN_CERTIFICATE
+    #   /// structure. The current revision level is 0x0200.
+    #   ///
+    #   UINT16  wRevision;
+    #   ///
+    #   /// The certificate type. See WIN_CERT_TYPE_xxx for the UEFI
+    #   /// certificate types. The UEFI specification reserves the range of
+    #   /// certificate type values from 0x0EF0 to 0x0EFF.
+    #   ///
+    #   UINT16  wCertificateType;
+    #   ///
+    #   /// The following is the actual certificate. The format of
+    #   /// the certificate depends on wCertificateType.
+    #   ///
+    #   /// UINT8 bCertificate[ANYSIZE_ARRAY];
+    #   ///
+    # } WIN_CERTIFICATE;
+    #
+    # #define WIN_CERT_TYPE_EFI_GUID         0x0EF1
+    #
+    # ///
+    # /// This identifies a signature containing a DER-encoded PKCS #7 version 1.5 [RFC2315]
+    # /// SignedData value.
+    # ///
+    # #define EFI_CERT_TYPE_PKCS7_GUID \
+    #   { \
+    #     0x4aafd29d, 0x68df, 0x49ee, {0x8a, 0xa9, 0x34, 0x7d, 0x37, 0x56, 0x65, 0xa7} \
+    #   }
+
+    _StructFormat = '<QIHH16s'
+    _StructSize = struct.calcsize(_StructFormat)
+
+    _MonotonicCountFormat = '<Q'
+    _MonotonicCountSize = struct.calcsize(_MonotonicCountFormat)
+
+    _StructAuthInfoFormat = '<IHH16s'
+    _StructAuthInfoSize = struct.calcsize(_StructAuthInfoFormat)
+
+    _WIN_CERT_REVISION = 0x0200
+    _WIN_CERT_TYPE_EFI_GUID = 0x0EF1
+    _EFI_CERT_TYPE_PKCS7_GUID = uuid.UUID('4aafd29d-68df-49ee-8aa9-347d375665a7')
+
+    def __init__(self):
+        self._Valid = False
+        self.MonotonicCount = 0
+        self.dwLength = self._StructAuthInfoSize
```

Comment:
this data is actually a wincert which we have defined
https://github.com/tianocore/edk2-pytool-library/blob/master/edk2toollib/uefi/wincert.py#L157